### PR TITLE
fix: reschedule task while socket is destroyed

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -143,11 +143,19 @@ class Client extends EventEmitter {
...
+    const dispatch = (request, cb) => {
       if (this.destroyed) {
         return cb(new Error('The client is destroyed'))
       }
 
+      if (this.socket.destroyed) {
+        // TODO: This doesn't work
+        // this[kQueue].unshift(request, cb)
+        // this[kQueue].pause()
+        // cb()
```
Comment (ronag) [inline lib/client.js:155]: @mcolline: I need help here... fastq is doing some magic here where it's not entirely clear how to reschedule a task.

**Comment by ronag** [comment]
@mcollina: This isn't pretty but solves the problem... do you have any better ideas?

File: lib/client.js [hunk 2]
```diff
@@ -143,11 +143,18 @@ class Client extends EventEmitter {
...
-    this[kQueue] = Q((request, cb) => {
+    const dispatch = (request, cb) => {
       if (this.destroyed) {
         return cb(new Error('The client is destroyed'))
       }
 
+      if (this.socket.destroyed) {
+        this[kQueue].unshift(request, cb)
+        this[kQueue].pause()
+        cb()
```
Comment (ronag) [inline lib/client.js:154]: The problem here is that we just want to reschedule the task without invoking the outer callback...

Comment (ronag) [reply lib/client.js:154]: Maybe something can be added to the fastq API in this regard?

**Comment by ronag** [comment]
This is kind of an edge case that might not be worth fixing...

File: lib/client.js (hunk 2 above)
Comment (mcollina) [reply lib/client.js:154]: take a look and see if it’s feasible

**Comment by ronag** [comment]
Won't be looking into this now, would be nice but not necessary

**Pull request closed without merging**